source/package.lisp

```diff
@@ -21,4 +21,8 @@
            #:*accessor-name-transformer*
            #:*automatic-accessors-p*
            #:*initarg-name-transformer*
-           #:*automatic-initargs-p*))
+           #:*automatic-initargs-p*
+           #:*slot-definition-transformer*
```

Comment:
these were never meant to be exported. they are only bound while the macroexpansion runs, so that these values don't need to be passed down through the chain of function calls that build up the final macroexpansion.

projects that extend defclass* itself can always :import-from the symbols that they want.

Reply:
But why exporting `*initarg-name-transformer*` and not `*slot-definition-transformer*` then?
Exporting these globals has the benefit of providing the user with customizable defaults, no? :)

Reply:
that's a good point, but exporting those was probably also a mistake.

it's something like readtime macros: while it's possible, it's impolite for a well behaving library to install readtime macros into the global readtable, interfering with all other packages loaded afterwards, let alone heisenbugs when the load order is random.

if someone has a toy, they can still access the private symbols if they want, but i wouldn't communicate to the users that changing these symbols is encouraged.

to resolve this i'm leaning towards unexporting those two also, not exporting the rest.

what do you think?

Reply:
Absolutely, this makes sense.

That said, it's nice to be able to set private defaults.  If I'm not mistaken, we could do this with metaclasses, right?  If so, I suggest we document and example of a metaclass with different defaults (e.g. exported class and accessor names).

Reply:
By the way, why making `defvar`s in the first place?  Couldn't we just leverage `build-defclass-like-expansion`?

Reply:
Friendly ping :)

Reply:
> By the way, why making `defvar`s in the first place? Couldn't we just leverage `build-defclass-like-expansion`?

a defvar, without a value, just declares the symbol special everywhere, both when referencing it, and when binding it.

Reply:
> If I'm not mistaken, we could do this with metaclasses, right?

some thing can be done with metaclasses, but using the MOP is decidedly out of scope for this project. the goal is to be a simple macro that generates a uniform boilerplate.

Reply:
Fair enough.  Shall we close this issue then?